### PR TITLE
8.8.0-pre

### PR DESCRIPTION
**Pull request by zliu41**

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Consensus.hs [hunk 1]
```diff
@@ -1335,7 +1335,10 @@ instance ( ToObject (ApplyTxErr blk), ToObject (GenTx blk),
   toObject verb (TraceMempoolRemoveTxs txs mpSz) =
     mconcat
       [ "kind" .= String "TraceMempoolRemoveTxs"
-      , "txs" .= map (toObject verb . txForgetValidated) txs
+      , "txs" .=
+        foldMap
+          (\(tx, err) -> [toObject verb (txForgetValidated tx), toObject verb err])
+          txs
```
Comment (amesgen) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Consensus.hs:1341]: Slightly more structured output:
```suggestion
          map
            (\(tx, err) -> Aeson.object
              [ "tx" .= toObject verb (txForgetValidated tx)
              , "reason" .= toObject verb err
              ])
            txs
```

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs [hunk 2]
```diff
@@ -867,7 +867,10 @@ instance
   forMachine dtal (TraceMempoolRemoveTxs txs mpSz) =
     mconcat
       [ "kind" .= String "TraceMempoolRemoveTxs"
-      , "txs" .= map (forMachine dtal . txForgetValidated) txs
+      , "txs" .=
+          foldMap
+            (\(tx, err) -> [forMachine dtal (txForgetValidated tx), forMachine dtal err])
+            txs
```
Comment (amesgen) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs:873]: As above:
```suggestion
          map
            (\(tx, err) -> Aeson.object
              [ "tx" .= forMachine dtal (txForgetValidated tx)
              , "reason" .= forMachine dtal err
              ])
            txs
```

**Review by amesgen — COMMENTED** [review]
The minor changes done here due to changes in Consensus seem good, only a minor comment :+1:

File: cabal.project [hunk 3]
```diff
@@ -32,8 +32,8 @@ packages:
   trace-resources
   trace-forward
 
-program-options
-  ghc-options: -Werror
+-- program-options
```
Comment (Jimbo4350) [inline cabal.project:35]: Need to remove `--`

File: cardano-node/cardano-node.cabal [hunk 4]
```diff
@@ -37,7 +37,7 @@ common project-config
                         -Wno-unticked-promoted-constructors
                         -Wpartial-fields
                         -Wredundant-constraints
-                        -Wunused-packages
+                        -- -Wunused-packages
```
Comment (Jimbo4350) [inline cardano-node/cardano-node.cabal:40]: Need to remove `--`

File: cardano-node/src/Cardano/Node/Tracing/Render.hs [hunk 5]
```diff
@@ -179,34 +179,38 @@ renderScriptIntegrityHash (Just witPPDataHash) =
...
-    Aeson.object [ "spending" .= Api.fromShelleyTxIn txin]
-  Alonzo.Rewarding rwdAcct ->
-    Aeson.object [ "rewarding" .= Aeson.String (Api.serialiseAddress $ Api.fromShelleyStakeAddr rwdAcct)]
-  Alonzo.Certifying _cert ->
-    Aeson.object
-      [ "certifying" .= Aeson.toJSON @String "TODO CIP-1694 unimplemented" -- toJSON (Api.textEnvelopeDefaultDescr $ Api.fromShelleyCertificate sbe cert)
-      ]
+renderScriptPurpose sbe plutusPurpose =
+  case sbe of
+    Api.ShelleyBasedEraShelley -> Aeson.Null
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Tracing/Render.hs:203]: `caseShelleyToMaryOrAlonzoEraOnwards` would be better suited here.

File: cardano-node/src/Cardano/Node/Tracing/Render.hs [hunk 6]
```diff
@@ -179,34 +179,38 @@ renderScriptIntegrityHash (Just witPPDataHash) =
...
+      case plutusPurpose of
+        AlonzoMinting pid ->
+          Aeson.object [ "minting" .= Aeson.toJSON pid]
+        AlonzoSpending (AsItem txin) ->
+          Aeson.object [ "spending" .= Api.fromShelleyTxIn txin]
+        AlonzoRewarding (AsItem rwdAcct) ->
+          Aeson.object [ "rewarding" .= Aeson.String (Api.serialiseAddress $ Api.fromShelleyStakeAddr rwdAcct)]
+        AlonzoCertifying _cert ->
+          Aeson.object
+            [ "certifying" .= Aeson.toJSON @String "TODO CIP-1694 unimplemented" -- toJSON (Api.textEnvelopeDefaultDescr $ Api.fromShelleyCertificate sbe cert)
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Tracing/Render.hs:214]: You also need to fill in the rest of the JSON rendering.

**Review by carbolymer — APPROVED** [review]

**Pull request merged**